packages/components/input/src/input.scss

```diff
@@ -1,58 +1,231 @@
+$sizes: md, lg;
+$variants: default, valid, invalid;
+$variant-selectors: ':not([valid],[invalid])', ':where([valid])', ':where([invalid])';
+
 :host {
-  --_border: 1px solid var(--_border-color);
-  --_border-color: #d3d3d3;
-  --_border-radius: 3px;
-  --_box-shadow: none;
-  --_gap: 0.25rem;
+  --_border-size: var(--sl-border-width-button-default);
+  --_border: var(--_border-size) solid var(--_border-color);
+  --_border-radius: var(--sl-border-radius-default);
+  --_box-shadow: var(--sl-box-shadow-none);
+  --_transition: 200ms ease-in-out;
 }
 
 :host {
-  align-items: start;
-  display: inline-flex;
-  flex-direction: column;
-  gap: var(--_gap);
+  display: block;
+  transition: var(--_transition);
   vertical-align: top;
+  width: 100%;
+}
+
+:host(:focus-visible) {
+  outline: none;
 }
 
 :host(:disabled) {
   cursor: default;
   pointer-events: none;
 }
 
-:host(:focus-within) {
-  --_border-color: hsl(207, 95%, calc(55% - 8% * 2));
-  --_box-shadow: 0 0 0 3px hsla(207, 95%, calc(55% - 8%), 0.25);
-}
-
-:host(:where(:--invalid, [state--invalid])) {
-  --_border-color: #c00;
-}
-
 slot[name='input']::slotted(input) {
   appearance: none;
   background: none;
   border: 0;
-  color: inherit;
+  color: var(--_color);
   flex: 1;
-  font: inherit;
-  line-height: 24px;
+  font: var(--_font);
   margin: 0;
   outline: none;
   padding: 0;
 
   &:focus {
     outline: none;
   }
+
+  &::placeholder {
+    color: var(--_placeholder-color);
+    opacity: 1;
+  }
 }
 
 .wrapper {
+  align-items: center;
+  background: var(--_background);
   border: var(--_border);
   border-radius: var(--_border-radius);
   box-shadow: var(--_box-shadow);
   box-sizing: border-box;
   display: flex;
-  padding: calc(0.25rem - 1px) 0.5rem;
-  transition: 0.2s ease-in-out;
+  gap: var(--_gap);
+  padding: var(--_padding);
+  transition: var(--_transition);
   transition-property: background, border, box-shadow, color, filter;
-  width: 100%;
+  width: auto;
+}
+
+.invalid-icon {
+  display: none;
+}
+
+:host(:where([invalid])) {
+  .invalid-icon {
+    display: block;
+  }
+}
+
+sl-error {
+  display: inline-flex;
+}
+
+/**
+ * theming
+ */
+
+@each $size in $sizes {
+  :host([size='#{$size}']) {
+    slot[name='input']::slotted(input) {
+      --_font: var(--sl-text-input-text-field-text-#{$size});
+    }
+
+    .wrapper {
+      --_padding: var(--sl-space-input-text-field-block-#{$size})
+        var(--sl-space-input-text-field-inline-#{$size})
+        var(--sl-space-input-text-field-block-#{$size})
+        var(--sl-space-input-text-field-inline-#{$size});
+      --_gap: var(--sl-space-input-text-field-gap-#{$size});
+    }
+  }
+}
+
+@each $variant in $variants {
```

Comment:
So in other stylesheets, I've moved this block of code to the top; between `:host {` and the actual component styling. So all the CSS variable stuff is grouped together.